workers/generic-worker/helper_test.go

```diff
@@ -304,14 +304,18 @@ func CreateArtifactFromFile(t *testing.T, path string, name string) (taskID stri
 		t.Fatalf("%#v", err)
 	}
 
-	// If task expires in the next two minutes, just fail intentionally. It
-	// isn't worth trying to handle this situation, since the task only expires
-	// after 6 months, so the chance of hitting the two minute period before it
-	// expires is extremely small, and the error will explicitly report it
-	// anyway.
+	// If task already expired but not purged from database, or expires in the
+	// next two minutes, just fail intentionally. It isn't worth trying to
+	// handle this situation, since the task only expires after 6 years, so the
+	// chance of hitting is reasonably small, and the error will explicitly
+	// report it anyway.
 	remainingTime := time.Until(time.Time(tdr.Expires))
 	if remainingTime.Seconds() < 120 {
-		t.Fatalf("You've been extremely unlucky. This test depends on task %q that was created six months ago but is due to expire in less than two minutes (%v). Wait a few minutes and try again!", taskID, remainingTime)
+		if remainingTime.Seconds() > 0 {
+			t.Fatalf("You've been extremely unlucky. This test depends on task %q that was created six years ago but is due to expire in less than two minutes (in %v). Wait until task purged and try again!", taskID, remainingTime)
```

Comment:
Expire tasks runs daily at 10am:
https://github.com/taskcluster/taskcluster/blob/62148cfcd47829ce4276a8e13f960343e6470006/services/queue/procs.yml#L25-L29

Maybe it would be useful to give more context in the error?